src/modules/NetworkLogger.ts

```diff
@@ -131,3 +161,87 @@ export const apolloLinkRequestHandler: RequestHandler = (operation, forward) =>
 
   return forward(operation);
 };
+
+/**
+ * @internal
+ * Exported for internal/testing purposes only.
+ */
+export const resetNetworkListener = () => {
+  if (process.env.NODE_ENV === 'test') {
+    _networkListener = null;
+  } else {
+    console.error(
+      `${InstabugConstants.IBG_APM_TAG}: The \`resetNetworkListener()\` method is intended solely for testing purposes.`,
+    );
+  }
+};
+
+/**
+ * @internal
+ * Exported for internal/testing purposes only.
+ */
+export const registerNetworkLogsListener = (
+  type: NetworkListenerType,
+  handler?: (networkSnapshot: NetworkData) => void,
+) => {
+  if (Platform.OS === 'ios') {
+    console.log('Andrew: registerNetworkLogsListener called');
+    // ignore repetitive calls
+    if (_networkListener === type || _networkListener === NetworkListenerType.both) {
+      console.log('Andrew: _registerNetworkLogsListener called on the same type');
+      return;
+    }
+    // remove old listeners
+    if (NetworkLoggerEmitter.listenerCount(NativeNetworkLoggerEvent.NETWORK_LOGGER_HANDLER) > 0) {
+      console.log('Andrew: removeAllListeners called');
+      NetworkLoggerEmitter.removeAllListeners(NativeNetworkLoggerEvent.NETWORK_LOGGER_HANDLER);
+    }
+
+    if (_networkListener == null) {
+      // set new listener.
+      _networkListener = type;
+    } else {
+      // attach a new listener to the existing one.
+      _networkListener = NetworkListenerType.both;
+    }
+    console.log(`Andrew: new NetworkLogsListener (${_networkListener}) attached`);
+  }
+
+  NetworkLoggerEmitter.addListener(
+    NativeNetworkLoggerEvent.NETWORK_LOGGER_HANDLER,
+    (networkSnapshot) => {
+      // Mapping the data [Native -> React-Native].
+      const { id, url, requestHeader, requestBody, responseHeader, response, responseCode } =
+        networkSnapshot;
+
+      // console.log(`Andrew: new snapshot ${url}`);
+      const networkSnapshotObj: NetworkData = {
+        id: id,
+        url: url,
+        requestBody: requestBody,
+        requestHeaders: requestHeader,
+        method: '',
+        responseBody: response,
+        responseCode: responseCode,
+        responseHeaders: responseHeader,
+        contentType: '',
+        duration: 0,
+        requestBodySize: 0,
+        responseBodySize: 0,
+        errorDomain: '',
+        errorCode: 0,
+        startTime: 0,
+        serverErrorMessage: '',
+        requestContentType: '',
+      };
```

Comment:
**comment**: I feel like with the new approach the obfuscation/filtering handlers don't get a `NetworkData` object anymore making this type misleading.

Reply:
We used the same type currently implemented in production to avoid breaking the existing API. I recall we previously discussed this and ultimately decided to proceed with the existing model.